From 5e28e486dbc793184ca462b8b528f5eff3d4f2ef Mon Sep 17 00:00:00 2001
From: jyasskin <jyasskin@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 1 Apr 2011 01:07:46 +0000
Subject: [PATCH] Merge r171807 from trunk to the 4.6 release branch, to fix two C++
 conformance problems in libsupc++.

Tested: bootstrap + make check-c++ on x86_64

libstdc++-v3/ChangeLog:

2011-03-31  Jeffrey Yasskin  <jyasskin@google.com>

        * libsupc++/exception_ptr.h: Forward-declare std::type_info.
        * libsupc++/nested_exception.h (__throw_with_nested): Remove a
	redundant default argument from std::__throw_with_nested.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@171815 138bc75d-0d04-0410-961f-82ee72b054a4

index ef826f6..26117cd 100644
--- a/libstdc++-v3/libsupc++/exception_ptr.h
+++ b/libstdc++-v3/libsupc++/exception_ptr.h
@@ -137,7 +137,7 @@ namespace std
       operator==(const exception_ptr&, const exception_ptr&) throw() 
       __attribute__ ((__pure__));
 
-      const type_info*
+      const class type_info*
       __cxa_exception_type() const throw() __attribute__ ((__pure__));
     };
 
diff --git a/libstdc++-v3/libsupc++/nested_exception.h b/libstdc++-v3/libsupc++/nested_exception.h
index 6a4f04e..d4804bb 100644
--- a/libstdc++-v3/libsupc++/nested_exception.h
+++ b/libstdc++-v3/libsupc++/nested_exception.h
@@ -117,7 +117,7 @@ namespace std
   // with a type that has an accessible nested_exception base.
   template<typename _Ex>
     inline void
-    __throw_with_nested(_Ex&& __ex, const nested_exception* = 0)
+    __throw_with_nested(_Ex&& __ex, const nested_exception*)
     { throw __ex; }
 
   template<typename _Ex>
-- 
1.7.0.4

